drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/page/audit/AuditPageViewImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.testscenario.client.page.audit;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import elemental2.dom.HTMLDivElement;
+import elemental2.dom.HTMLElement;
+import org.drools.workbench.models.testscenarios.shared.ExecutionTrace;
+import org.drools.workbench.screens.testscenario.client.resources.i18n.TestScenarioConstants;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Templated
+@Dependent
+public class AuditPageViewImpl implements AuditPage.AuditPageView {
+
+    @DataField("root")
+    private HTMLDivElement root;
+
+    @DataField("audit-log")
+    private HTMLDivElement auditLogDiv;
+
+    @DataField("fired-rules")
+    private HTMLDivElement firedRulesDiv;
+
+    private AuditPage presenter;
+
+    private AuditTable firedRulesTable;
+
+    private AuditTable auditLogTable;
+
+    @Inject
+    public AuditPageViewImpl(final HTMLDivElement root,
+                             final HTMLDivElement auditLogDiv,
+                             final HTMLDivElement firedRulesDiv,
+                             final AuditTable firedRulesTable,
+                             final AuditTable auditLogTable) {
+        this.root = root;
+        this.auditLogDiv = auditLogDiv;
+        this.firedRulesDiv = firedRulesDiv;
+        this.firedRulesTable = firedRulesTable;
+        this.auditLogTable = auditLogTable;
+
+        this.firedRulesDiv.appendChild(firedRulesTable.getElement());
+        this.auditLogDiv.appendChild(auditLogTable.getElement());
+
+        this.auditLogTable.setTitle(TestScenarioConstants.INSTANCE.AuditLog());
+        this.firedRulesTable.setTitle(TestScenarioConstants.INSTANCE.FiredRules());
+    }
+
+    @Override
+    public HTMLElement getElement() {
+        return root;
+    }
+
+    @Override
+    public void init(AuditPage presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void showFiredRules(final ExecutionTrace executionTrace) {
+        firedRulesTable.showItems(Arrays.asList(executionTrace.getRulesFired()));
+    }
+
+    @Override
+    public void showAuditLog(final Set<String> auditLogMessages) {
+        auditLogTable.showItems(auditLogMessages);
```

Comment:
IDK if the use of ```Set``` has changed at all in this refactoring; however I'd be inclined to convert to a ```List``` and sort it before passing to the View for display. WDYT?

Reply:
I think this is not good way, the data looks like [1]. I think if we want to improve audit log, we should return proper list directly form backend.

[1]
![screenshot from 2018-01-24 22-58-20](https://user-images.githubusercontent.com/8044780/35359399-3d77b9d8-015a-11e8-97c4-2bbbf9c03521.png)

Reply:
Ok. Fair enough.